core/tests/app/test_app.py

```diff
@@ -440,12 +451,81 @@ def test_set_empty_full_screen_window_list(event_loop):
     # Change the screens that are full screen
     app.set_full_screen(window1, window2)
     assert app.is_full_screen
-    assert_action_performed_with(app, "enter_full_screen", windows=(window1, window2))
-
+    assert_action_performed_with(
+        app,
+        "enter presentation mode",
+        screen_window_dict={app.screens[0]: window1, app.screens[1]: window2},
+    )
     # Exit full screen mode by setting no windows full screen
     app.set_full_screen()
     assert not app.is_full_screen
-    assert_action_performed_with(app, "exit_full_screen", windows=(window1, window2))
+    assert_action_performed(
+        app,
+        "exit presentation mode",
+    )
+
+
+def test_presentation_mode(event_loop):
+    """The app can be put into presentation mode."""
+    app = toga.App(formal_name="Test App", app_id="org.example.test")
+    window1 = toga.Window()
+    window2 = toga.Window()
+
+    assert not app.is_in_presentation_mode
+
+    # If we're not in presentation mode, exiting presentation mode is a no-op
```

Comment:
Rather than one long test, it's better to have a large number of parameterised short tests - that way you can validate which specific pieces of behavior are failing, rather than getting a single "it's broken" test failure.

Reply:
I have broken the tests of presentation mode into smaller tests, but they are not parameterized. This because, doing:
```
@pytest.mark.parametrize(
    "windows",
    [
        [],
        [toga.Window()],
        [toga.Window(), toga.Window()],
        [toga.Window(), toga.Window(), toga.Window()],
        {},
        {toga.App.app.screens[0]: toga.Window()},
        {
            toga.App.app.screens[0]: toga.Window(),
            toga.App.app.screens[1]: toga.Window(),
        },
    ],
)
def test_presentation_mode_with_windows_list(event_loop, app, windows):
```
gives error, since the app is not instantiated. I have tried several variations of the above, but all of them gave error because of the app not being instantiated. I also found out that I cannot use pytest fixtures like `app` fixxture inside `@pytest.mark.parametrize` decorator.

Hence, I have instead broken down presentation mode test into smaller tests.

Reply:
I have thought of another way to achieve parameterization of the test without creating additional tests, by:
```
@pytest.mark.parametrize(
    "num_of_windows, data_type",
    [
        (0, list),
        (1, list),
        (2, list),
        (3, list),
        (0, dict),
        (1, dict),
        (2, dict),
    ],
)
def test_presentation_mode_with_windows_list(event_loop, app, num_of_windows, data_type):
```
Let me know if I should change the current core tests by the above paramterized test.

Reply:
In general, I'd advise away from this as an approach. Tests should have as little logic in them as possible - because otherwise, you've got additional logic that needs to be tested. Sometimes logic is unavoidable; but if the option is *slightly* more verbose code or logic, go with verbose code.

Also - when picking test cases, be wary of what it is you're actually testing. What does validating the 2 window case tell you that 3 windows doesn't? Unless there's a good reason why the 2 and 3 case is different, there's no need to include both.